### PR TITLE
Notifications Initializer Migration to Pack tag

### LINKED ISSUE #18998: Migrate Notification initializer to Pack tag
In an effort to migrate away from asset pipeline, the current initializers need to be migrated to pack tags via Webpacker and initialized in the perspective view/page.

The `initNotifications()` should be migrated to a pack tag and added to the perspective views in which this code should be initialized.

### PR DESCRIPTION
**Pull request by lboogie04**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Related Tickets & Documents

- Related Issue #
- Closes #18998 

## QA Instructions, Screenshots, Recordings

1. Log in with a user that should have notifications(Notifications may have to be created manually)
2. Upon logging in, the notification bell should STILL show the `notification_count` on the notification bell in the upper right corner of the nav.
3. Proceed to the `notifications` page
4. Try saving and liking a post.
5. Also try loading more posts at the bottom if possible.

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/javascript/utilities/sendHapticMessage.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+      window.webkit &&
+      window.webkit.messageHandlers &&
+      window.webkit.messageHandlers.haptic
+    ) {
+      window.webkit.messageHandlers.haptic.postMessage(message);
+    }
+  } catch (err) {
+    console.log(err.message); // eslint-disable-line no-console
+  }
+}
```
Comment (lightalloy) [inline app/javascript/utilities/sendHapticMessage.js:14]: Missing newline )

**Comment by Ridhwana** [comment]
Hi @lboogie04 👋 Are there any QA instructions that you can recommend for testing the migration of this functionality?

**Comment by Ridhwana** [comment]
Hey @lboogie04  😊 Just wanted to check in on this again in case you missed my [comment] (https://github.com/forem/forem/pull/19124#issuecomment-1484786761)? If you'd prefer though I could go ahead and review based only on the code and not pull up the UI?

File: app/javascript/packs/initializers/initializeNotifications.js [hunk 2]
```diff
@@ -108,16 +101,16 @@ function initReactions() {
...
+            .getElementById(`comment-form-for-${  thisButt.dataset.reactableId}`)
             .classList.remove('hidden');
           thisButt.classList.add('hidden');
           thisButt.classList.remove('inline-flex');
-          setTimeout(function () {
+          setTimeout(() => {
             document
               .getElementById(
-                'comment-textarea-for-' + thisButt.dataset.reactableId,
+                `comment-textarea-for-${  thisButt.dataset.reactableId}`,
```
Comment (Ridhwana) [inline app/javascript/packs/initializers/initializeNotifications.js:113]:
```suggestion
                `comment-textarea-for-${ thisButt.dataset.reactableId }`,
```

File: app/javascript/packs/initializers/initializeNotifications.js [hunk 3]
```diff
@@ -108,16 +101,16 @@ function initReactions() {
 
         butt.onclick = function (event) {
           event.preventDefault();
-          var thisButt = this;
+          const thisButt = this;
           document
-            .getElementById('comment-form-for-' + thisButt.dataset.reactableId)
+            .getElementById(`comment-form-for-${  thisButt.dataset.reactableId}`)
```
Comment (Ridhwana) [inline app/javascript/packs/initializers/initializeNotifications.js:106]:
```suggestion
            .getElementById(`comment-form-for-${ thisButt.dataset.reactableId }`)
```

File: app/javascript/packs/initializers/initializeNotifications.js [hunk 4]
```diff
@@ -198,3 +191,13 @@ function initLoadMoreButton() {
     button.addEventListener('click', initPagination);
   }
 }
+
+export function initializeNotifications() {
+  fetchNotificationsCount();
```
Comment (Ridhwana) [inline app/javascript/packs/initializers/initializeNotifications.js:196]: I'm wondering if some of this code should be covered by cypress tests, this function almost reads like a user experience from start to finish hence it sounds a little interaction heavy and may possibly be a good candidate for an end to end test if it isn't covered already. What are your thoughts @lboogie04 ?

**Review by Ridhwana — COMMENTED** [review]
It looks great @lboogie04! Just left a question for you to get your feedback.

**Comment by rt4914** [comment]
Will review this PR tomorrow. Thanks.

**Review by jaw6 — APPROVED** [review]
Followed QA steps locally, looked like everything worked as expected :+1:

**Review by Ridhwana — APPROVED** [review]
Thanks for adding the tests and Qa instructions @lboogie04! LGTM 🎉

**Pull request merged**